### ISSUE TITLE
Kaggle silently compresses some archives

### ISSUE THREAD

**Issue by 20zinnm** [tensorflow/datasets#844]
**Short description**
Kaggle seems to silently compress large files which causes errors when trying to access the downloaded files at the paths given by the download manager.

**Environment information**
* Operating System: Windows 10
* Python version: 3.6.8
* `tensorflow-datasets` version: 1.1.0
* `tensorflow-gpu` version: 2.0.0-beta1

**Reproduction instructions**

Join the Severstal Steel Defect detection competition and add:

```python
dl_manager.download_kaggle_data('severstal-steel-defect-detection')
```

to any existing builder. Notice that Kaggle downloads `train.csv` as a ZIP archive despite reporting its file name without `.zip`.

**Expected behavior**
Tensorflow Datasets should check that Kaggle was honest with its file name. This might need to happen in `KaggleCompetitionDownloader.download_file()` after line 137.

```python
if not fname.endswith('.zip') and tf.io.gfile.exists(os.path.join(output_dir, fname + '.zip')):
  # Kaggle silently compressed the file.
  # Decompress it here?
```

**Comment by 20zinnm** [comment]
This behavior is inconsistent, too--Kaggle CLI does not compress some files. Also, when running `kaggle competitions download -c severstal-steel-defect-detection` from the command line, it properly identifies the file as `train.csv.zip`.

**Comment by us** [comment]
Kaggle downloader works like that:

`kaggle competitions --file {filenames} download -c severstal-steel-defect-detection`

Can you try this?

**Comment by 20zinnm** [comment]
@us I just confirmed that the above command (after moving `--file {filenames}` to after `download`) results in a file `train.csv.zip` being downloaded. Note that I specified `train.csv` for the filename; Kaggle automatically compressed it before downloading.

That's a problem because the tfds Kaggle downloader does not properly account for the possibility that some files will be compressed automatically while others will not.
